src/ResourceManagement/Graph.RBAC/CertificateCredentialImpl.cs

```diff
@@ -140,11 +140,12 @@ internal CertificateCredentialImpl(KeyCredential keyCredential)
             }
         }
 
-        internal CertificateCredentialImpl(string name, IHasCredential<T> parent, string customKeyIdentifier)
+        internal CertificateCredentialImpl(string name, IHasCredential<T> parent, string customKeyIdentifier, Guid? keyId)
             : base(name, new KeyCredential()
             {
                 Usage = "Verify",
                 CustomKeyIdentifier = customKeyIdentifier,
+                KeyId = (keyId ?? Guid.NewGuid()).ToString(),
```

Comment:
So previous KeyId is null. And service would generate something for it upon PUT?

Reply:
Yeah, the service will generate a UUID as KeyID, if it is null. The azure-cli generate keyId by itself, so I think this behavior would not be changed. https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/role/custom.py#L1604